pkg/scheduler/queue/tree_queue_algo_querier_worker_queue_priority.go

```diff
@@ -4,74 +4,54 @@ package queue
 
 // QuerierWorkerQueuePriorityAlgo implements QueuingAlgorithm by mapping worker IDs to a queue node to prioritize.
 // Querier-workers' prioritized queue nodes are calculated by the integer workerID % len(nodeOrder).
+// This distribution of workers across query component subtrees ensures that when one query component is experiencing
+// high latency about 25% of querier-workers continue prioritizing queries for unaffected components.
 //
-// Purpose:
-// This algorithm is intended to ensure that querier-workers are balanced across nodes in the queue selection tree.
-// While it can serve to balance consumers across request queue nodes which have been partitioned by any criteria,
-// its original purpose is to balance querier-workers across queues partitioned a query's expected query component`.
-//
-// There are four possible query components: "ingester", "store-gateway", "ingester-and-store-gateway", and "unknown".
-// With all four possible query component queue nodes active, the modulo operation on the worker IDs
-// will distribute approximately 1 / 4 of the querier-workers to each queue node.
-// A querier-worker ID which is mapped to prioritize the "ingester" queue node will always start there
-// and attempt first to work on a query which only requires the ingesters to complete.
-//
-// By splitting the queues by query component we can ensure that the approximately 1 / 4 of querier-workers are
-// "reserved" for a query component even when the other query component is experiencing high latency.
-// This reservation ensures that the querier-worker connections can continue to process queries
-// which do not need to utilize the query component experiencing high latency.
-//
-// Performance:
-// This significantly outperforms the previous round-robin algorithm which simply rotated through the node order
-// (see TestMultiDimensionalQueueAlgorithmSlowConsumerEffects benchmark outputs for comparison).
+// This significantly outperforms the previous round-robin approach which simply rotated through the node order.
 // Although a vanilla round-robin algorithm will select a given query-component node 1 / 4 of the time,
-// in situations of high latency on a query component that one slow query component will still grow asymptotically
+// in situations of high latency on a query component, the one slow query component will still grow asymptotically
```

Comment:
```suggestion
// in situations of high latency on a query component, the queue size of one slow query component will still grow asymptotically
```

Reply:
I updated to this:

```
// in situations of high latency on a query component, the utilization of the querier-worker connections
// as measured by inflight query processing time will grow asymptotically to be dominated by the slow query component.
```

I am trying to express that the round robin still allows the slow query component to grow to use all connections:
With the slow component only being 8x slower than the fast component - taking 8 time slices to process instead of 1, by time slice 10, all consumers are processing the slow queries, and we will spend the rest of the time essentially bouncing between 0 and 1 consumers processing fast queries.

|  | round-robin selection state | consumer 1 | consumer 2 | consumer 3 | consumer 4 |
| :---- | :---- | :---- | :---- | :---- | :---- |
| time 1 | fast | dequeue fast (advance to slow) |  |  |  |
| time 2 | slow | process fast | dequeue slow (advance to fast) |  |  |
| time 3 | fast | wait | process slow | dequeue fast (advance to slow) |  |
| time 4 | slow | wait | process slow | process fast | dequeue slow (advance to fast) |
| time 5 | fast | dequeue fast (advance to slow) | process slow | wait | process slow |
| time 6 | slow | process fast | process slow | dequeue slow (advance to fast) | process slow |
| time 7 | fast | dequeue fast (advance to slow) | process slow | process slow | process slow |
| time 8 | slow | process fast | process slow | process slow | process slow |
| time 9 | slow | dequeue slow (advance to fast) | process slow | process slow | process slow |
| time 10 | fast | process slow | process slow | process slow | process slow |
| time 11 | fast | process slow | dequeue fast (advance to slow) | process slow | process slow |
| time 12 | slow | process slow | process fast | process slow | process slow |
| time 13 | fast | process slow | wait | process slow | dequeue slow (advance to fast) |
| time 14 | fast | process slow | dequeue fast (advance to slow) | process slow | process slow |
| time 15 | slow | process slow | process fast | dequeue slow (advance to fast) | process slow |
| time 16 | fast | process slow | dequeue fast (advance to slow) | process slow | process slow |
|  |  | process slow | process fast | process slow | process slow |
|  |  |  |  | process slow | process slow |
|  |  |  |  | process slow | process slow |
|  |  |  |  | process slow | process slow |
|  |  |  |  | process slow | process slow |
|  |  |  |  | process slow |  |
|  |  |  |  | process slow |  |
|  |  |  |  |  |  |
|  |  |  |  |  |  |